src/Triangle.cpp

```diff
@@ -95,34 +95,39 @@ float Triangle::compute_t(Vector source,Ray r)
     return t;
 }
 
-bool Triangle::ray_intersect_plan(Ray r,Point& P)
+Point Triangle::ray_intersect_plan(Ray r,float t)
 {	//A*x + B*y + C*z + D = 0
     //A B C = Normal
     //D = distance from the origin (0,0,0)
     //D = N * P0 where P0 point of triangle
     //Step 1 : Finding P
     //P = Source + t*Dir
 
-    Point O(0,0,0);
-    Point S = r.get_source();
-    Vector source(O,S);
-    double t = compute_t(source,r);
+    Vector source(Point(0, 0, 0),r.get_source());
```

Comment:
A cleaner way would be to create a function in the point class that takes a vector and translates the point by a vector
```suggestion
    Vector source(Point(0, 0, 0),r.get_source());
    Vector V = source +t*r.get_direction();         //Compute point of intersect with plan
    return Point(V.get_x(),V.get_y(),V.get_z());
```